egi_notebooks_hub/egispawner.py

```diff
@@ -152,62 +150,82 @@ class DataHubSpawner(EGISpawner):
-                            "class": "onedatafs_jupyter.OnedataFSContentsManager",
-                            "config": {"space": "/" + datahub_response["name"]},
-                        }
-                    )
-                except HTTPError as e:
-                    self.log.info("Something failed! %s", e)
-                    raise e
-            pod.spec.containers[0].args = pod.spec.containers[0].args + [
-                ("--NotebookApp.contents_manager_class=%s" % self.manager_class),
-                (
-                    "--OnedataFSContentsManager.oneprovider_host=$(%s)"
-                    % self.oneprovider_env
-                ),
-                ("--OnedataFSContentsManager.access_token=$(%s)" % self.token_env),
-                ('--OnedataFSContentsManager.path=""'),
-                ("--OnedataFSContentsManager.force_proxy_io=%s" % self.force_proxy_io),
-                (
-                    "--OnedataFSContentsManager.force_direct_io=%s"
-                    % self.force_direct_io
-                ),
-                ("--MixedContentsManager.filesystem_scheme=%s" % json.dumps(scheme)),
-            ]
-            self.log.info("POD: %s", pod.spec.containers[0].args)
-        return pod
+    oneprovider_storage_mapping = List(
+        [],
+        config=True,
+        help="""
+        List of dicts like:
+            {"storage_id": "<oneprovider storage id>",
+             "mount_point": "volume mount point"}
+        """,
+    )
+
+    extra_mounts = List([], config=True, help="""extra volume mounts in k8s""")
+
+    async def pre_spawn_hook(self, spawner):
+        host = spawner.environment.get("ONEPROVIDER_HOST", "")
+        token = spawner.environment.get("ONECLIENT_ACCESS_TOKEN", "")
+        cmd = ["oneclient", "-f"]
+        cmd.append(f"-H {host}")
+        if self.force_proxy_io:
+            cmd.append("--force-proxy-io")
+        if self.force_direct_io:
+            cmd.append("--force-direct-io")
+        if self.oneprovider_storage_mapping:
+            for mapping in self.oneprovider_storage_mapping:
+                cmd.append(
+                    "--override %(storage_id)s:mountPoint:%(mount_point)s" % mapping
+                )
+        cmd.append(self.mount_point)
+        volume_mounts = [
+            {"mountPath": f"{self.mount_point}:shared", "name": "oneclient"},
+        ]
+        if self.extra_mounts:
+            volume_mounts.extend(self.extra_mounts)
+        spawner.extra_containers = [
+            {
+                "name": "oneclient",
+                "image": self.client_image,
+                "env": [
+                    {"name": "ONECLIENT_PROVIDER_HOST", "value": host},
+                    {"name": "ONECLIENT_ACCESS_TOKEN", "value": token},
+                ],
+                "resources": {
+                    "requests": {"memory": "512Mi", "cpu": "250m"},
+                    "limits": {"memory": "4Gi", "cpu": "500m"},
```

Comment:
maybe better to have lower limits, i hope we are not going to use 4GB per sidecar container

Reply:
made this configurable